### PR TITLE
Handle non-ok responses

### PR DESCRIPTION
**Pull request by dmfalke**
If a request for static content results in a response that is not `ok`, show an error message where the content would be displayed.

### CODE REVIEW

File: Site/webapp/wdkCustomization/js/client/components/homepage/FeaturedTools.tsx [hunk 1]
```diff
@@ -32,20 +33,31 @@ type FeaturedToolEntry = {
...
-        const response = await fetch(`https://${communitySiteUrl}${FEATURED_TOOL_URL_SEGMENT}`, { mode: 'cors' });
-
-        // FIXME Validate this JSON using a Decoder
-        const responseData = await response.json() as FeaturedToolResponseData;
-
-        setFeaturedToolResponseData(responseData);
+        try {
+          const response = await fetch(`https://${communitySiteUrl}${FEATURED_TOOL_URL_SEGMENT}`, { mode: 'cors' });
+          if (response.ok) {
+            const responseData = await response.json() as FeaturedToolResponseData;
```
Comment (jtlong3rd) [inline Site/webapp/wdkCustomization/js/client/components/homepage/FeaturedTools.tsx:51]: We should carry over this FIXME from the previous implementation, as you did for `WorkshopExercises`:

`// FIXME Validate this JSON using a Decoder`

**Review by jtlong3rd — COMMENTED** [review]
Looks good to me! I had one small inline comment.

File: Site/webapp/wdkCustomization/js/client/components/homepage/FeaturedTools.tsx (hunk 1 above)
Comment (dmfalke) [reply Site/webapp/wdkCustomization/js/client/components/homepage/FeaturedTools.tsx:51]: Oops. Good catch!

**Comment by dmfalke** [comment]
Thanks! I'll update shortly.

File: Site/webapp/wdkCustomization/js/client/components/homepage/FeaturedTools.tsx (hunk 1 above)
Comment (dmfalke) [reply Site/webapp/wdkCustomization/js/client/components/homepage/FeaturedTools.tsx:52]: I put the comment back.

**Review by jtlong3rd — APPROVED** [review]

**Pull request merged**